lib/openstack_query/query_parser.py

```diff
@@ -44,7 +32,10 @@ def parse_sort_by(self, *sort_by: Tuple[PropEnum, bool]) -> None:
         """
         self._sort_by = {}
         for prop, order in sort_by:
```

Comment:
Nit: Do this before the for loop

```
if (any user_selected_prop not in self._prop_enum_cls for user_selected_prop in sort_by):
  raise ParseQueryError(
                    f"Error: Given property to select: {prop.name} is not supported by query"
                )
if any (group_by not in self._prop_enum_cls for group_by in sort_by):
  raise ...

for prop,order in sort_by:
   ...
```

This keeps our processing and pre-condition logic separate